frontends/halo2/src/adaptors.rs

```diff
@@ -0,0 +1,58 @@
+//! Adaptor traits that clients need to implement for communicating with the [`crate::driver::Driver`].
+
+use ff::Field;
+
+use crate::lookups::LookupData;
+
+/// Boxed constraints system adaptor.
+pub type CSA<F> = Box<dyn ConstraintSystemAdaptor<F>>;
+
+/// Trait for querying information about the constraint system derived during configuration.
+pub trait ConstraintSystemAdaptor<F: Field> {
+    /// Returns the list of gates defined in the system.
+    fn gates(&self) -> Vec<&dyn GateAdaptor<F>>;
+
+    /// Returns a list with data about the lookups defined in the system.
+    fn lookups<'cs>(&'cs self) -> Vec<LookupData<'cs, F>>;
+}
+
+/// Temporary implementation of [`ConstraintSystemAdaptor`].
+impl<F: Field> ConstraintSystemAdaptor<F> for halo2_proofs::plonk::ConstraintSystem<F> {
```

Comment:
In general I think Adaptors are classes that are meant to glue two systems. If this is mainly an interface to query a constraint system then let's call it a `ConstraintSystemInfoManager` or `CSInfoManager`